packages/monaco/src/browser/textmate/monaco-theme-registry.ts

```diff
@@ -0,0 +1,115 @@
+
+/********************************************************************************
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { IRawTheme, Registry } from 'monaco-textmate';
+
+export interface ThemeMix extends IRawTheme, monaco.editor.IStandaloneThemeData {}
+
+export class MonacoThemeRegistry {
+
+    protected themes = new Map<string, ThemeMix>();
+
+    public getTheme(name: string): IRawTheme| undefined {
+        return this.themes.get(name);
+    }
+
+    public register(json: any, includes?: {[includePath: string]: any}, givenName?: string, monacoBase?: monaco.editor.BuiltinTheme): ThemeMix {
```

Comment:
Just wondering if the registry should convert every theme like this?

This parser is only meant for vscode-like themes, but is this the only format we should accept?